agent/websockets/shim.go

```diff
@@ -272,12 +273,14 @@ func ShimBody(shimPath string) (func(resp *http.Response) error, error) {
 type sessionMessage struct {
 	ID      string      `json:"id,omitempty"`
 	Message interface{} `json:"msg,omitempty"`
+	Version int         `json:"v,omitempty"`
 }
 
 func createShimChannel(ctx context.Context, host, shimPath string, rewriteHost bool, openWebsocketWrapper func(http.Handler) http.Handler) http.Handler {
 	var connections sync.Map
 	var sessionCount uint64
 	mux := http.NewServeMux()
+	var protocolVersion int
```

Comment:
You have this as a single, mutable variable in the proxy agent that gets updated every time a new client opens up a websocket-shim connection.

However, I don't think that will work, as different clients might be speaking different versions of the protocol.

Instead, I would recommend adding this as a field of the `Connection` struct. That way the protocol version will be determined separately for each new websocket-shim connection based on the `open` request sent by the client, and that version will remain consistent for as long as the connection is open.

Reply:
Good catch! Done.